src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Caching.Memory;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;img&gt; elements that supports file versioning.
+    /// </summary>
+    /// <remarks>
+    /// The tag helper won't process for cases with just the 'src' attribute.
+    /// </remarks>
+    [TargetElement("img", Attributes = FileVersionAttributeName)]
+    public class ImageTagHelper : TagHelper
+    {
+        private static readonly string Namespace = typeof(ImageTagHelper).Namespace;
+
+        private const string FileVersionAttributeName = "asp-file-version";
+        private const string SrcAttributeName = "src";
+
+        private FileVersionProvider _fileVersionProvider;
+        
+        /// <summary>
+        /// Source of the image.
+        /// </summary>
+        /// <remarks>
+        /// Passed through to the generated HTML in all cases.
+        /// </remarks>
+        [HtmlAttributeName(SrcAttributeName)]
+        public string Src { get; set; }
```

Comment:
> since it's a `string` a `null` value wont be possible if its in `TargetElement`.

Side note: Put that way, it sounds like a bug.  Razor doesn't do what author likely meant if author writes `src="@null"` for a bound `string` attribute.

But I agree having both `SrcAttributeName` and `FileVersionAttributeName` in the `TargetElement.Attributes` value makes sense.  Avoids some useless executions of the helper.